org.civicrm.reqangular/src/common/services/pub-sub.js

```diff
@@ -0,0 +1,59 @@
+define([
+  'common/modules/services'
+], function (services) {
+  'use strict';
+
+  var topics = Object.create(null);
+
+  /**
+   * This service is used for communication between different angular apps using pub-sub pattern
+   * @ngdoc service
+   * @name pubSub
+   * @requires $rootScope
+   **/
+
+  services.factory("pubSub", ['$rootScope', function ($rootScope) {
+    return {
+      /**
+       * Subscribe method for events
+       * @ngdoc method
+       * @name subscribe
+       * @methodOf pubSub
+       * @returns Object
+       */
+      subscribe: function (topic, listener) {
+        var index;
+        topics[topic] = topics[topic] || [];
+        listener.$rootScope = $rootScope;
+        index = topics[topic].push(listener) - 1;
+
+        return {
+          remove: function () {
+            delete topics[topic][index];
+          }
+        };
+      },
+
+      /**
+       * Publish method for events
+       * @ngdoc method
+       * @name publish
+       * @methodOf pubSub
+       * @returns void
+       */
+      publish: function (topic, data) {
+        if (!topics[topic]) {
+          return;
+        }
+        data = data != undefined ? data : {};
+
+        topics[topic].forEach(function (listener) {
+          listener(data);
+          listener.$rootScope.$applyAsync();
```

Comment:
Can you use `$applyAsync` the standard way?
```js
listener.$rootScope.$applyAsync(function () {
  listener(data);
});
```
both `$apply` and `$applyAsync` are expecting an expression to evaluate passed as a param